lib/util.js

```diff
@@ -430,3 +431,150 @@ module.exports = exports = {
                   'util.puts is deprecated. Use console.log instead.',
                   'DEP0027')
 };
+
+if (process.binding('config').experimentalReport) {
+  try {
+    // if node-report is enabled, extract the binding and
+    // wrap the APIs with thin layers, with some error checks.
+    // user options can come in from CLI / ENV / API.
+    // CLI and ENV is intercepted in C++ and the API call here (JS).
+    // So sync up with both sides as appropriate - initially from
+    // C++ to JS and from JS to C++ whenever the API is called.
+    // Some events are controlled purely from JS (signal | exception)
+    // and some from C++ (fatalerror) so this syncup is essential for
+    // correct behavior and aligned to the supplied tunables.
+    const nr = internalBinding('node_report');
+
+    var report = {
```

Comment:
`const report = {`

Reply:
the problem is that I as we are very liberal with the `setDiagnosticReportOptions` in terms of the input (users can chose to pass all or none of the input) we want to `clean` this object before we parse the input, else the stale entries set in the previous call be be still in effect. So that leads to needing to modify this object, and hence `var`.

However, given the vulnerability of user interference as reported you (next comment), I have moved this outside of the export bundle, and kept it as a private `var config`. Hope this adjustment is fine with you?